sklearn/tree/export.py

```diff
@@ -850,7 +851,8 @@ def export_text(decision_tree, feature_names=None, max_depth=10,
     """
     check_is_fitted(decision_tree)
     tree_ = decision_tree.tree_
-    class_names = decision_tree.classes_
+    if is_classifier(decision_tree):
```

Comment:
why not in line 898 where there's already an ``if``?

Reply:
there is an attribute error if we access `decision_tree.classes_` on a regressor